osbs/build/build_requestv2.py

```diff
@@ -0,0 +1,225 @@
+        :param platform: str, platform
+        :param koji_target: str, koji tag with packages used to build the image
+        :param koji_task_id: str, koji ID
+        :param koji_parent_build: str,
+        :param flatpak: if we should build a Flatpak OCI Image
+        :param flatpak_base_image: str, name of the Flatpack OCI Image
+        :param reactor_config_map: str, name of the config map containing the reactor environment
+        :param yum_repourls: list of str, uris of the yum repos to pull from
+        :param signing_intent: bool, True to sign the resulting image
+        :param compose_ids: list of str,
+        :param filesystem_koji_task_id: int, Koji Task that created the base filesystem
+        :param platform_node_selector: dict, a nodeselector for a user_paramsific platform
+        :param scratch_build_node_selector: dict, a nodeselector for scratch builds
+        :param explicit_build_node_selector: dict, a nodeselector for explicit builds
+        :param auto_build_node_selector: dict, a nodeselector for auto builds
+        :param isolated_build_node_selector: dict, a nodeselector for isolated builds
+        :param additional_tag_data: dict, str of dir path and filename of additional tags
+                                    and set of str of tags
+        """
+
+        # Here we cater to the koji "scratch" build type, this will disable
+        # all plugins that might cause importing of data to koji
+        self.scratch = kwargs.get('scratch')
+        # When true, it indicates build was automatically started by
+        # OpenShift via a trigger, for instance ImageChangeTrigger
+        self.is_auto = kwargs.pop('is_auto', False)
+        # An isolated build is meant to patch a certain release and not
+        # update transient tags in container registry
+        self.isolated = kwargs.get('isolated')
+
+        self.validate_build_variation()
+
+        self.base_image = kwargs.get('base_image')
+        self.platform_node_selector = kwargs.get('platform_node_selector', {})
+        self.scratch_build_node_selector = kwargs.get('scratch_build_node_selector', {})
+        self.explicit_build_node_selector = kwargs.get('explicit_build_node_selector', {})
+        self.auto_build_node_selector = kwargs.get('auto_build_node_selector', {})
+        self.isolated_build_node_selector = kwargs.get('isolated_build_node_selector', {})
+
+        logger.debug("setting params '%s' for %s", kwargs, self.user_params)
+        self.user_params.set_params(**kwargs)
+
+    # Override
+    @property
+    def inner_template(self):
+        raise RuntimeError('inner_template not supported in BuildRequestV2')
+
+    # Override
+    @property
+    def customize_conf(self):
+        raise RuntimeError('customize_conf not supported in BuildRequestV2')
+
+    # Override
+    @property
+    def dj(self):
+        raise RuntimeError('DockJson not supported in BuildRequestV2')
+
+    def set_reactor_config_map(self, api):
```

Comment:
No need to override. See parent's `set_reactor_config`.

Come to think of it, should that method move to this class? Usage of `reactor_config_map/override` is really only meant to be supported in arrangement 6.

Reply:
set_reactor_config_map() existed in BuildRequest before this latest round of changes. I'm not sure about removing it.

I'll just add the parameters to BuildRequest.set_reactor_config() and BuildRequest.set_required_secrets() and see how that goes.